packages/mattermost-redux/src/actions/users.ts

```diff
@@ -113,57 +114,60 @@ export function loadMe(): ActionFunc {
         const serverVersion = state.entities.general.serverVersion || Client4.getServerVersion();
         dispatch(setServerVersion(serverVersion));
 
-        let responseData: CurrentUserInfoQueryResponseType['data'] | null = null;
+        let userId: UserProfile['id'] = '';
+        let userRolesAsString = '';
+
         try {
-            const {data} = await Client4.fetchWithGraphQL<CurrentUserInfoQueryResponseType>(currentUserInfoQuery);
-            responseData = data;
-        } catch (error) {
-            dispatch(logError(error));
-            return {error};
-        }
+            const {data, errors} = await Client4.fetchWithGraphQL<CurrentUserInfoQueryResponseType>(currentUserInfoQuery);
```

Comment:
Not necessarily something that needs to be changed, but because `fetchWithGraphQL` just returns an array of error objects, it's a bit different than how the rest of the `Client4` methods work because those all throw for network errors. That's something I've considered changing because I like Go's error handling style, but it does require changing a lot of how we do error handling like how you had to throw a new error below this and then use type assertions to force it to be a ServerError in the catch statement.

Anyway, that's just something to think about since it might be only a one-off thing that we're doing here. If we end up having to rethrow errors a lot or do a lot of type assertions, that might be a sign that we could be doing something better though.

Reply:
with graphQL its different because for errors it will still return "200" and hence that catch.

Reply:
I see. I guess we could make `fetchWithGraphQL` throw when that happens to match the other Client4 methods. That way, we could also have it only return `data` so that we don't have to destructure the result to get that value

Reply:
Yes, thats a good idea. made a follow up ticket https://mattermost.atlassian.net/browse/MM-48332